File: app/src/main/java/org/mesonet/app/androidsystem/UserSettings.java
package org.mesonet.app.androidsystem;

import android.content.Context;
import android.content.SharedPreferences;

import javax.inject.Inject;


public class UserSettings
{
    private static final String kMesonetSettings = "MesonetSettings";
    public static final String kUnitPreference = "UnitPreference";
    public static final String kSelectedStid = "SelectedStid";
    public static final String kSelectedRadar = "SelectedRadar";

    @Inject
    UserSettings(){}




    public void SetPreference(Context inContext, String inName, String inValue)
    {
        SharedPreferences preferences = inContext.getSharedPreferences(kMesonetSettings, Context.MODE_PRIVATE);

        SharedPreferences.Editor editor = preferences.edit();
        editor.putString(inName, inValue);
        editor.apply();
    }



    public String GetStringPreference(Context inContext, String inName, String inDefault)
    {
        SharedPreferences preferences = inContext.getSharedPreferences(kMesonetSettings, Context.MODE_PRIVATE);

        String result = preferences.getString(inName, inDefault);
        return result;
    }
}

File: app/src/main/java/org/mesonet/app/androidsystem/UserSettings.kt
package org.mesonet.app.androidsystem

import android.content.Context

import javax.inject.Inject


class UserSettings @Inject
internal constructor() {


    fun SetPreference(inContext: Context, inName: String, inValue: String) {
        val preferences = inContext.getSharedPreferences(kMesonetSettings, Context.MODE_PRIVATE)

        val editor = preferences.edit()
        editor.putString(inName, inValue)
        editor.apply()
    }


    fun GetStringPreference(inContext: Context, inName: String, inDefault: String): String {
        val preferences = inContext.getSharedPreferences(kMesonetSettings, Context.MODE_PRIVATE)

        return preferences.getString(inName, inDefault)
    }

    companion object {
        private val kMesonetSettings = "MesonetSettings"
        val kUnitPreference = "UnitPreference"
        val kSelectedStid = "SelectedStid"
        val kSelectedRadar = "SelectedRadar"
    }
}
